metrics/exporter_test.go

```diff
@@ -260,7 +263,7 @@ func TestFlushExporter(t *testing.T) {
 		reportingPeriod:    1 * time.Minute,
 		backendDestination: Prometheus,
 	}
-	e, err := newMetricsExporter(c, TestLogger(t))
+	e, _, err := newMetricsExporter(c, TestLogger(t))
```

Comment:
we don't seem to have tests for `factory` return value.

Reply:
Good point, I'll try to add that.

Reply:
The `resource_view_test.go` implicitly tests the factory value; I'm trying to figure out (other than asserting that the function is non-nil) what a useful test of the factory value would look like short of the integration test.

Reply:
I can check that it's not nil here, but the tests in `resource_view_test.go` seem much more useful and likely to test the actual functionality.

Reply:
Always verify it's not `nil`?

Reply:
Done.